src/accelerate/utils/other.py

```diff
@@ -77,8 +77,7 @@ def extract_model_from_parallel(model, keep_fp32_wrapper: bool = True, recursive
     """
     options = (torch.nn.parallel.DistributedDataParallel, torch.nn.DataParallel)
 
-    is_compiled = is_compiled_module(model)
```

Comment:
To clarify: Is `is_compiled` being assigned here and then checked after DDP/DS is unwrapped? If so, this seems  potentially unnecessary or wrong.

Reply:
Yes I agree that we should check for compilation after

Reply:
Sorry I think I misunderstood the code. `is_compiled` is assigned here before unwrapping to keep the torch compilation. I still think this could be a problem if the model is in the form of `Distributed(Compiled(model))`. Refer to my newest commit.